staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -29,11 +29,13 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 )
 
-// The short keepalive timeout and interval have been chosen to aggressively
-// detect a failed etcd server without introducing much overhead.
 var (
+	// The short keepalive timeout and interval have been chosen to aggressively
+	// detect a failed etcd server without introducing much overhead.
 	keepaliveTime    = 30 * time.Second
 	keepaliveTimeout = 10 * time.Second
+	// dialTimeout is the timeout for failing to establish a connection.
+	dialTimeout = 10 * time.Second
```

Comment:
@kubernetes/sig-scalability-pr-reviews - Do you have any stats on a sane default on establishing a connection on a loaded cluster?  10 seems fine to me, but just want to be sure. 

/cc @jpbetz

Reply:
@timothysc @jpbetz We don't have any stats on dial timeout, but 10-second seems fine and is better than none. If dial timeout is not specified, `grpc.DialContext` would block on unavailable nodes for initial connection creation.

We are more interested in finding out optimal values for keepalives (default in k8s https://github.com/kubernetes/kubernetes/pull/58008/files).

Reply:
we should better expose this keepalive config as a follow up.